### PR TITLE
[Test] Nightly shuffle test

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

@krfricke Is this going to be automatically available to the nightly test framework once it is merged? 

Dask on Ray tests are in progress 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dask_on_ray/dask_on_ray_app_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+base_image: "anyscale/ray-ml:pinned-nightly"
+env_vars: {}
+debian_packages: []
+
+python:
+  pip_packages: [dask, tqdm, fastparquet, numpy==1.19.5, scipy, xarray, dataclasses, zarr]
+  conda_packages: []
+
+post_build_cmds:
+  - pip install -U "ray[default]@https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl"
```
Comment (krfricke) [inline release/nightly_tests/dask_on_ray/dask_on_ray_app_config.yaml:10]:
```suggestion
  - pip3 install -U {{ env["RAY_WHEELS"] | default("ray") }}"
  - pip3 install -U ray[default]
```

File: release/nightly_tests/shuffle/shuffle_app_config.yaml [hunk 2]
```diff
@@ -6,4 +6,7 @@ python:
   pip_packages: []
   conda_packages: []
 
-post_build_cmds: []
+post_build_cmds:
+  - pip uninstall -y ray
+  - pip install -U "ray[default]@https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl"
```
Comment (krfricke) [inline release/nightly_tests/shuffle/shuffle_app_config.yaml:11]:
```suggestion
  - pip3 install -U {{ env["RAY_WHEELS"] | default("ray") }}"
  - pip3 install -U ray[default]
```

File: release/nightly_tests/dask_on_ray/large_scale_test.py [hunk 3]
```diff
@@ -0,0 +1,456 @@
+"""
```
Comment (krfricke) [inline release/nightly_tests/dask_on_ray/large_scale_test.py:1]: Is this file currently used anywhere? Should it be added to the tests? (some test yaml?)

**Review by krfricke — CHANGES_REQUESTED** [review]

File: release/nightly_tests/nightly_tests.yaml [hunk 4]
```diff
@@ -1,15 +1,72 @@
...
+  run:
+    timeout: 3000
+    prepare: python wait_cluster.py 4 600
+    script: python shuffle/shuffle_test.py --num-partitions=200 --partition-size=500e6
+
+  smoke_test:
+    cluster:
+      compute_template: shuffle/shuffle_compute_smoke.yaml  # Does not exist yet
+
+# Test multi nodes 100GB shuffle with large partitions
```
Comment (ericl) [inline release/nightly_tests/nightly_tests.yaml:56]:
```suggestion
# Test multi nodes 100GB shuffle with large number of partitions
```

File: release/nightly_tests/nightly_tests.yaml [hunk 5]
```diff
@@ -1,15 +1,72 @@
...
+
+  run:
+    timeout: 3000
+    script: python shuffle/shuffle_test.py --num-partitions=500 --partition-size=50e6
+
+  smoke_test:
+    cluster:
+      compute_template: shuffle/shuffle_compute_smoke.yaml  # Does not exist yet
+
+# Test multi nodes 100GB shuffle with small partitions
```
Comment (ericl) [inline release/nightly_tests/nightly_tests.yaml:37]:
```suggestion
# Test multi nodes 100GB shuffle with small number of partitions
```

File: release/nightly_tests/nightly_tests.yaml [hunk 6]
```diff
@@ -1,15 +1,72 @@
...
-    timeout: 600
-    script: python shuffle/shuffle_10gb.py
+    timeout: 3000
+    script: python shuffle/shuffle_test.py --num-partitions=50 --partition-size=200e6
+
+  smoke_test:
+    cluster:
+      compute_template: shuffle/shuffle_compute_smoke.yaml  # Does not exist yet
+
+# Test single node 10GB shuffle with large partitions.
```
Comment (ericl) [inline release/nightly_tests/nightly_tests.yaml:19]:
```suggestion
# Test single node 10GB shuffle
```

File: release/nightly_tests/dask_on_ray/large_scale_test.py (hunk 3 above)
Comment (rkooo567) [reply release/nightly_tests/dask_on_ray/large_scale_test.py:1]: I will add it after making sure shuffle test results are reported correctly. (in the followup PR)

File: release/data_processing_tests/multi_node.yaml [hunk 7]
```diff
@@ -108,7 +108,8 @@ setup_commands:
     # - sudo apt install -y gdb
     # - sudo apt install -y linux-tools-common
     - echo 'export PATH="$HOME/anaconda3/envs/tensorflow_p36/bin:$PATH"' >> ~/.bashrc
-    - pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/test_wheels/latest/ray-2.0.0.dev0-cp36-cp36m-manylinux2014_x86_64.whl
+    - pip uninstall -y ray
+    - pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp36-cp36m-manylinux2014_x86_64.whl
```
Comment (krfricke) [inline release/data_processing_tests/multi_node.yaml:112]: This is a non e2e release test, right? As in, it is started manually and uses the Ray OSS autoscaler?

**Review by krfricke — APPROVED** [review]
Generally looks good to me, just a final question

File: release/data_processing_tests/multi_node.yaml (hunk 7 above)
Comment (rkooo567) [reply release/data_processing_tests/multi_node.yaml:112]: It is not now. I will do that once shuffle test results are properly reported.

**Comment by rkooo567** [comment]
All of the build failure should be unrelated to this PR as it only changes the testing folder.

**Pull request merged**